src/exchange/group/delegation/group-delegation.controller.js

```diff
@@ -103,17 +124,46 @@ angular.module('Module.exchange.controllers').controller(
         });
     }
 
+    fetchAccountCreationOptions() {
+      this.loadingDomains = true;
+      return this.services.Exchange.fetchingAccountCreationOptions(
+        this.$routerParams.organization,
+        this.$routerParams.productId,
+      )
+        .then((accountCreationOptions) => {
+          const domains = [this.allDomainsOption];
+          _.forEach(accountCreationOptions.availableDomains, (domain) => {
+            domains.push(domain);
+            if (domain.name === this.selectedDomain.name) {
```

Comment:
You can then `find` the value

Reply:
Yes, I can use a _.map or array concatenation and then _.find to find the selected item. I believe this will traverse the array twice, once by _.map and once by _.find. This will take O(N) + O(N) time complexity. I if do both inside a forEach, the time taken will be O(N). Please read my above comment and let me know If I am wrong.